### ISSUE TITLE
Update resolution is not working

### ISSUE THREAD

**Issue by saoron** [dride/dride-app-issues#4]
Steps to reproduction:

Update a resolution, save and the setting will go to default..

**Comment by saoron** [comment]
Issue resolved! Will be released in a new firmware update.

**Comment by saoron** [comment]
Anyone wants to test it before the public release?

**Comment by BilliTheKid** [comment]
Yes I'd like to test it out please. I'm on Android

**Comment by saoron** [comment]
Please put this file on the SD card (Unzipped) and turn on your Dride, It will blink red and eventually turn on white white, Then re-insert the SD to your computer and remove the file. Done.

[FW96663A.bin.zip](https://github.com/dride/dride-app-issues/files/3755725/FW96663A.bin.zip)


Let me know if it's working for you..

**Comment by BilliTheKid** [comment]
Hi Yossi. 

I have been at it for about 35 mins and the zip file does not sort the issue with the resolution saving. It still defaults back to the 1280 setting. All other settings within the app saves but the resolution.

I placed the zip file on to the root of the card. Placed it in the unit and it flickered white. Connected via the app try to change settings and nothing. I then removed the file from the SD card and inserted it and tried again and it's still the same I'm afraid.

**Comment by saoron** [comment]
You should see red light flickering, that means it's upgrading the FW, Can you post a pic of your SD root dir?

**Comment by JerryBlack** [comment]
Following the instructions from saoron (which worked exactly as described) I can confirm that the FW96663A.bin  doesn't show the current setting in the pick list and doesn't save a newly chosen recording resolution. (It would be nice if the app also reported the firmware version as well as the app version)

**Comment by BilliTheKid** [comment]
So it seems it hasn't worked. :-(

**Comment by saoron** [comment]
Could you try this one please?

[FW96663A.bin.zip](https://github.com/dride/dride-app-issues/files/3757284/FW96663A.bin.zip)

**Comment by BilliTheKid** [comment]
Here is a picture of the root of the SD card with the zip file
![Screenshot_20191022-180751](https://user-images.githubusercontent.com/56843472/67311124-e6091d80-f4f6-11e9-86d9-e892218c846c.png)

**Comment by JerryBlack** [comment]
The unzipped file should be copied to the card.

**Comment by BilliTheKid** [comment]
Yes I can confirm working now after you unzip the file and put the bin file on the root of the directory. It was stated to not unzip it but you must and it has worked and the settings save successfully. Bravo!

**Comment by MyDefender** [comment]
Does it fix any other issues, e.g. having to reset the Dride before being able to connect to it via the app?

**Comment by BilliTheKid** [comment]
I don't have those issues. Power it off turn it on, open the app and it connects.

**Comment by MyDefender** [comment]
Lucky you.

**Comment by BilliTheKid** [comment]
Timestamp still out by a day and the time is still wrong but it fixes the saving settings for sure

**Comment by BilliTheKid** [comment]
The only thing I've noticed is that when it's unlpugged /no power to the unit. It beeps several times ever so often. Is that normal?

**Comment by saoron** [comment]
@BilliTheKid we're working on that two issues, they will be updated in their tickets. 
The beeping is for shutting down and booting up, Do you get them in a raw?

**Comment by BilliTheKid** [comment]
When I switch it off as in no power going to the unit the beeps start. And they do it around 5 times with about 8/10 seconds of silence before another sets of beeps. Yes in RAW.

**Comment by JerryBlack** [comment]
For me, the 2nd FW96663A.bin did not help with displaying the resolution from the pick list. ( I didn't do a paper clip reset). When I change the resolution to 2560x1440@30 I see that reflected in the settings menu, but when I pop-up the selection list a 2nd time, it isn't pre-checked. Also if I leave the settings menu and go back in, it isn't displayed on the settings page (no value is shown for resolution). That makes it hard to tell if it is set (unlike the impact detection and sound, which I can set and see the current value).  The only visible change is that the dride wifi is now shown as CarDV-xxx (proof that the update worked?)    The date is also incorrect 21:15 vs my current time of 10:27 (at the moment). I would like to access the drive from my MacBook, previously I could connect to the dride wifi and it would let me browse at http://192.168.1.254/DCIM/MOVIE.  Currently that doesn't work. My MacBook (which uses DHCP with a manual address 192.168.2.50 ). Is it the manual addressing that is the issue? Is there a better address for the base dride?  Thanks.

**Comment by saoron** [comment]
@BilliTheKid You need to remove the firmware file before you use your Dride, or else it will run FW update in a loop.

**Comment by saoron** [comment]
@JerryBlack Is your app updated to the latest version?

**Comment by JerryBlack** [comment]
I’m on 5.6.2   is that the latest?

**Comment by BilliTheKid** [comment]
Yes I did that. All is goiod and working

**Comment by Mensra** [comment]
I followed the instructions in the GitHub thread for resolution change not "sticking" and now my dride zero shows CarDV-6246 for wifi, default P/W. I can navigate to the root directory in a web browser. The dride app now acts like there is no device, tells me to go to settings and connect to dride wifi, but I AM connected, see above.  It seems that the device is recording videos, but unless I remove the card, there is no way to see them. Unit is worthless in current state. I installed the GitHub firmware since I was only able to connect to the dash cam once, when it was first installed. I am using the hardwire kit, professionally installed. Please advise, I hate to have spent the money for the install, I can always throw away the dride and buy one from another mfg that actually works, but am willing to wait a bit. app version 5.6.2 on iOS 13.2 beta 4 iPhone X

**Comment by magier76** [comment]
Well i did the update and Dride blinked red and then turned white.
When i check the firmware version nothing is changed.....
Time settings are oke now, but still not able to change the resolution.
I'm using a Iphone X iOS 13.2
Dride firmware: MTN_190906_0.0.1

**Comment by saoron** [comment]
Updated firmware is out guys, could you please test this out?

[FW96663A.bin.zip](https://github.com/dride/dride-app-issues/files/3793878/FW96663A.bin.zip)

**Comment by Mensra** [comment]
Do you have a change log?

David

**Comment by Mensra** [comment]
Now my video is upside down. How can I flip it?

**Comment by Mensra** [comment]
Download full resolution images
Available until Nov 30, 2019
 <https://www.icloud.com/attachment/?u=https%3A%2F%2Fcvws.icloud-content.com%2FB%2FAWbQxqTCXgupwCQvaTBXuip_w1WaAWPT68QltwalwVAUL4SCzhPFWGDe%2F%24%7Bf%7D%3Fo%3DAqTqkk1YJoYmvf1eJL8iVIEAFhVr_EX4K1EPEIL6iKLE%26v%3D1%26x%3D3%26a%3DCAogv7NQwLKeIzP9iMJzy4Jke6RfDf2bEJ668zYsHk35jy0SJxCFrsiU4i0Yhb7D6OstIgEAKggByAD_C3NkFFIEf8NVmloExVhg3g%26e%3D1575130029%26k%3D%24%7Buk%7D%26fl%3D%26r%3D66FFB421-0D10-4733-A2A4-1289178C99ED-1%26ckc%3Dcom.apple.largeattachment%26ckz%3D151E6D92-9B60-4C5E-A0F4-4221C34493F1%26p%3D33%26s%3Dkq8_tXxPG0VaB175-BSLGdwXQIE&uk=f2mK5FFEkjXkbz9v-CyrcQ&f=Images.zip&sz=292775>My unit has video upside down and the Led goes white and then red. Stays red. Does not want too reconnect reliably.
Here is what is on my SD card.

> MS-DOS (FAT32)

**Comment by Mensra** [comment]
Here is the memory card I’m using:



David

**Comment by JerryBlack** [comment]
OK, updated using the firmware from 4 hours ago, I was at MTN_190906_0.0.1, and now am at DrideZero_V2.0.0  
1) The resolution is now shown on the initial setup screen (1280x720 60fps) and it wasn't before (for me). When selecting a new value from the pick list, the new chosen value is checked. After exiting the setup and returning, the same (1280x720 60fps) is still showing.  
2) In addition, now my front camera image is now upside down (it wasn't before). Choosing Flip image slider has no effect and doesn't appear to stick either.
3) After the update, now my LED is both white and red, so obviously it's not happy about this either :)
Cheers

**Comment by saoron** [comment]
Thanks for checking this out,

1. After FW update the settings will be defaulted, Anyone else have the same experience as @JerryBlack ?
2. Image flipped - this is expected, im looking for someone that doesnt get this, anyone?
3. You need to remove the firmware from the SD card after it finishes blinking red.

**Comment by JerryBlack** [comment]
I did remove the firmware from the SD card previously. I just went through (my version of) the install a 2nd time and I'm left with both a white/red LED again (with the .bin removed). Just how long does an update take and what are the colours/sounds indicating the update sequence and completion so that I know when to remove the SD card?    Mine was  sold red, blinking red, white/red.

**Comment by Mensra** [comment]
I have the exact same status. So we have to leave our camera with the image flip? When will you have a correction firmware update?

**Comment by Mensra** [comment]
Now the camera is beeping at me and the white light has a red tinge to it. I can no longer connect to the camera. I have taken the card out again and re-formatted it, nothing works. The camera apparently is recording but it’s beeping regularly. I have pulled the power out to let it run down because I don’t want to drive with something beeping in my car. Either get this thing working, or give me my money back and the money I spent to install it. It has cost me more than if I just went out and bought a blackvue 4K dual cam for $500.

**Comment by BilliTheKid** [comment]
> Updated firmware is out guys, could you please test this out?
> 
> [FW96663A.bin.zip](https://github.com/dride/dride-app-issues/files/3793878/FW96663A.bin.zip)

What is the difference between this and the one released 17 days ago?

**Comment by JerryBlack** [comment]
This version appears to have solved most of my problems I think at 1st glance, although the firmware version provides the same number MTN_190906_0.0.1. The app is at version 5.7.  One thing different about the install was that the dride had been left unplugged overnight (rather than accessory fuse power) before using this, so perhaps there may have been a positive side effect from cold booting with the fresh install. Even the time might be correct (as long as you aren't displaying time in 24 hour format). I will post a new comment after further use.

**Comment by BilliTheKid** [comment]
@JerryBlack Does it fix the inverted image and timestamp?

**Comment by JerryBlack** [comment]
@BilliTheKid @saoron This last update fixed the inverted image. The time stamp is correct most of the time, some of the earlier clips viewed when scrolling down to earlier ones, appear to have the wrong timestamp, but I can't see a pattern.  Still no ability to alter the video resolution and see an increase in the menu. I set the resolution up to 2540 and waited for a new video (given the menu doesn't indicate the current resolution). This video when downloaded to dropbox is still only 1280 x 720 (I presume the upload to dropbox isn't auto downsizing the resolution). Also Deleting saved clips from the main window (...  delete on the image) has no permanent effect. They disappear, but come back if you relaunch the app.

**Comment by BilliTheKid** [comment]
@JerryBlack @saoron  So there has been 2 zip file firmware updates right? The only issue i had was the timestamp which was tiny, incorrect and lots of random numbers and letters and the resolution not changing when saved. When I done the first update it fixed that resolution issue but the time-stamp is still incorrect. Not sure I want to flash this one if the resolution is not working again.

**Comment by BilliTheKid** [comment]
@JerryBlack @saoron As you can see the timestamp in mine is perfect. No invert with camera and can select the various resolutions and it saves too no bother.
![Screenshot_20191106-194116](https://user-images.githubusercontent.com/56843472/68331974-da5c5000-00cd-11ea-9a29-f8440d32984b.png)
![Uploading Screenshot_20191106-194116.png…]()

**Comment by BilliTheKid** [comment]
Have a video of me changing resolutions and saving but it won't allow me upload on this

**Comment by BilliTheKid** [comment]
Sorry I stand corrected. As you can see the November 6th was correct and then the next one taken afterwards is November 7th which is wrong so definitely still a bug there.
![Uploading Screenshot_20191106-194818.png…]

**Comment by BilliTheKid** [comment]
![Screenshot_20191106-194818](https://user-images.githubusercontent.com/56843472/68332421-b0575d80-00ce-11ea-9620-a047fd219f1c.png)

**Comment by BilliTheKid** [comment]
@saoron So what is the latest firmware out beyond the one you have me to flash 15 days ago? Is there a newer one to fix the time-stamp as I don't seem to have any further issues as I can see. @JerryBlack

**Comment by BilliTheKid** [comment]
The only other issue which was mentioned before is when you unplug the cable and give it no power the LED comes on an glows white then goes off and makes a noise. Does this about 3 times. Again I have a video but unable to upload it here @saoron

**Comment by BilliTheKid** [comment]
![Screenshot_20191106-200404](https://user-images.githubusercontent.com/56843472/68333550-a0407d80-00d0-11ea-93aa-e4bf84063278.png)

**Comment by BilliTheKid** [comment]
![Screenshot_20191106-200343](https://user-images.githubusercontent.com/56843472/68333586-b0f0f380-00d0-11ea-9324-771c3f7bfda7.png)

**Comment by JerryBlack** [comment]
@BilliTheKid @saoron I misspoke in my last message. The timestamp is off (by almost a day). Apparently today isn't Nov. 7th.

**Comment by BilliTheKid** [comment]
@JerryBlack @saoron Ah ok. Strange thing is. The first video i showed you in the picture the time stamp was correct but everyone after that is out stating it's November 7th. Do you have the issue of when it's unplugged that it chimes a few times when it has no power?

**Comment by JerryBlack** [comment]
@BilliTheKid I have my mounted in the vehicle with continuous power, so I haven't noticed that.

**Comment by BilliTheKid** [comment]
@JerryBlack @saoron As you can see it's displayed at 2560 x 1440 so the clip is recording at the higher resolution but it's not 4K.
QHD refers to 2560 x 1440
![Screenshot_20191106-205512](https://user-images.githubusercontent.com/56843472/68337293-15fc1780-00d8-11ea-9e0f-582191701e3f.png)

**Comment by BilliTheKid** [comment]
@JerryBlack @saoron Mmmm. I haven't installed mine as I'm still awaiting a 10m mini USB to be delivered as I don't want to use the cigarette charger as I only have USB's in my car. However, when I turn off the engine then there is no power going to the unit so I'd imagine it will suffer with these chimes from the unit with no power going to it. As you can see from the photos above the white light goes on for a while then off and I get a chime. Happens about 5/6 times before it finally shutdown completely. So got to be a bug.

**Comment by JerryBlack** [comment]
@BilliTheKid I ordered mine with the fuse kit (or it came as part of being an early supporter). I had to buy fuse taps to piggy back into the fuse box (note, there are 3 different sizes of taps, Subaru Outbacks take the smallest format I found by trial and error).

**Comment by BilliTheKid** [comment]
@JerryBlack I didn't go down the fuse box route. What sold me was the promo video showing the mini USB cable (which believe it or not is the hardest cable to buy in-store as practically no tech product uses this connection anymore) Was disappointed to not find the cable advertised in that PROMO video in the box. Not a deal breaker, but they should of went with a more ready available connecter. It's actually the lowest drawing power connector compared to micro and type C. So been waiting for that cable to arrive before installing but prefer to be conducting the fixes without it installed. So hopefully the fix that @saoron states is coming will squash all the remaining bugs and I'll be able to enjoy my Dride eventually.

**Comment by saoron** [comment]
Guys, I've just uploaded a new beta, would be great if you could test it out..

**Comment by MyDefender** [comment]
I can’t find the new beta on here. Where is it please Yossi?

**Comment by saoron** [comment]
https://github.com/dride/dride-app-issues/releases

**Comment by BilliTheKid** [comment]
@saoron What has been fixed in this beta?

**Comment by saoron** [comment]
Yes..

**Comment by BilliTheKid** [comment]
No what has been fixed in this beta as there is no log. Can you tell me?

**Comment by BilliTheKid** [comment]
@saoron Just flashed the beta version. Resolution is working and so is the time stamp. Only issue I can see from this firmware is the image has been inverted/upside down. Is there a way to fix this?

**Comment by Mensra** [comment]
I will flash the next beta that does not have any known issues. If the image is inverted, that is a known issue which will prevent this from even being theoretically functional. Beta firmware is firmware that’s being tested for obscure bugs, not with known issues preventing them from being ever used. 

David

**Comment by BilliTheKid** [comment]
@saoron Also it seems to have fixed the disconnection issue too of it not shutting off correctly so all good. So remaining bugs what I can see is the deleting issue of videos not deleting correctly, the camera been inverted and the issue with mobile data not working when connected to Dride, but things looking good and getting there

**Comment by BilliTheKid** [comment]
@Mensra All betas contain issues hence why they are beta's

**Comment by Mensra** [comment]
@BilliTheKid, actually beta is for testing to discover unknown issues. We are talking about is alpha to test general functionality. The purpose of beta testing is to thrash out problems that are not obvious.
See also: https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=21&ved=2ahUKEwim2LrP6N_lAhWymeAKHY74DSMQFjAUegQIBRAB&url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FSoftware_testing&usg=AOvVaw1kqyhgI3l6I7_Ax8ceXIO1

**Comment by BilliTheKid** [comment]
Isn't that what I said? 😂

**Comment by Mensra** [comment]
Not exactly. Beta software is to be released when you think you have tested it and covered all the obvious major flaws. Inverted image is an obvious (non-hidden) flaw, and should not be in a beta test release. You beta test to find the hidden problems, not problems that you know have to be fixed. Very often if beta tested software passes the real world testing, it becomes a final version. Clearly, this particular version has a major, non–obvious issue and that should not have been released for beta testing. No if he had designated this as testing for a particular function, and not as a beta test, I would’ve agreed with his designation. I have beta tested many products, both hardware and software. A fair number of them have been so good that the beta tested product went directly to golden master. Beta testing is particularly important for complex systems that may interact in unpredictable manners for outlier conditions.

**Comment by saoron** [comment]
Guys could you please keep this technical? It's been overblown already, it makes it hard to follow.  This types of discussions are welcome on the forum.

**Comment by JerryBlack** [comment]
The latest beta six hours ago does invert the image for me, the lights remain white and red, I am unable to delete images. The time is wrong. The resolution is displayed but changes don’t stick. Would you like us to upload log files?

**Comment by BilliTheKid** [comment]
@JerryBlack Mmm. Resolution change back and forth is fine for me. My LED is solid white as it should be. Only issues I can see is the inverted image, individual videos not deleting properly although when you click delete all video's it does it then starts recording again.

**Comment by JerryBlack** [comment]
@BilliTheKid did you see any change in firmware number? I presume that this wasn’t a firmware update. The only visible indicator that I have successfully updated the system Are: the wifi name reverted to carDV.... , the resolution shows, the settings were reset, the images are upside down, the light is white +red.

**Comment by JerryBlack** [comment]
@BilliTheKid do you have a rear camera? My front image is inverted, the rear image is OK

**Comment by BilliTheKid** [comment]
@JerryBlack Did you flash the new beta?

**Comment by JerryBlack** [comment]
@BilliTheKid good point, it turns out I may have uploaded an OLD version. I can’t check/retry until about 8 hours from now.

**Comment by BilliTheKid** [comment]
@JerryBlack @saoron Change logs should be highlighted for each firmware release. I flashed it and it went smooth only for the issues i stated above so they are nearly there

**Comment by BilliTheKid** [comment]
@JerryBlack Did you update to the latest beta?

**Comment by JerryBlack** [comment]
@BilliTheKid I believe so (hard to tell :) 
My current issues are: 
1) live view of the front camera is inverted (stored video is OK),
2) resolutions show, but changing the resolution doesn't stick,
3) Light shows red for a bit, and then shows white(left side)/red(right side) unit functions [maybe the bin file wasn't removed, but I thought I did that].
4) Interesting in that viewing the images through the app, only video segments when the car was parked are visible. Video of driving between 2 times where the car was parked, don't show in the display of videos (power issue from the fuse box?) whereas they used to.

what works now: 
5) I do have speed showing now.
6) I can record 3 minute videos.
7) Date/time is correct
8) Other settings work I think.

Cheers

**Comment by BilliTheKid** [comment]
@JerryBlack 

So to answer that.

live view of the front camera is inverted (stored video is OK),

Yes same

resolutions show, but changing the resolution doesn't stick,

Mine does but not sure if the resolution is recording in the set resolution.

Light shows red for a bit, and then shows white(left side)/red(right side) unit functions [maybe the bin file wasn't removed, but I thought I did that].

No issues here in that.

Interesting in that viewing the images through the app, only video segments when the car was parked are visible. Video of driving between 2 times where the car was parked, don't show in the display of videos (power issue from the fuse box?) whereas they used to.

Not sure on that one as I'm using USB

what works now:
4) I do have speed showing now.

Where is that viewable?

5) I can record 3 minute videos.

Set to 3 minute but it still only records for 30 seconds maybe not sure why.

6) Date/time is correct

Yes same.

7) Other settings work I think.

Haven't tested it all but what I have noticed aswell is that when sharing an image via email, WhatsApp the compression that is used is too harsh especially set at a high resolution.

@saoron

**Comment by JerryBlack** [comment]
@BilliTheKid 
4) the speed is at the bottom of the video to the right of the date.

**Comment by BilliTheKid** [comment]
@JerryBlack Ah right. I actually haven't used it in my car as of yet would you believe. I'm still awaiting my long length mini USB to arrive to install it properly so wouldn't have noticed the speed. Prefer to wait until all the bugs are squashed before doing so. It's great that @saoron is addressing the bugs in frequent firmware and app updates but it's a shame that when 1 bug is squashed another appears.

**Comment by saoron** [comment]
Hi Guys,
I've just released a new beta, would appreciate if you could take it for a spin.

https://github.com/dride/dride-app-issues/releases/tag/2.0-beta-1

**Comment by JerryBlack** [comment]
@saoron some progress with this new version. 
1. The live view inversion has been fixed for me. 
Outstanding issues:
1. Setting an alternate (higher) resolution doesn't stick, i.e. changing works, but if you exit the setup mode and go back in, it reports the default (I don't have the app in front of me now, but it's the bottom item 1280x720@60?).
2. The 1st couple of videos after (/during) the update process have the time wrong (in the future) whereas those immediate preceding the update were correct. I haven't checked to see whether it has corrected itself since.

I am getting a white/red light combo, which doesn't seem right. What is that an indication of?

Thanks

**Comment by BilliTheKid** [comment]
Will test now and report back @saoron @JerryBlack

**Comment by BilliTheKid** [comment]
@saoron @JerryBlack 

OK so here are my findings.

1. The image has corrected itself.
2. When i save a resolution setting and exit the app and go back in it is still locked in place.
3. When you go into settings and press delete all video's it deletes them.
4. When I want to delete individual video's it deletes them and they do not reappear after I have closed the app.
5. Boot up time of connecting from the app to the unit is a little slower in this firmware update.
6. Can browse internet and use data while connected to Dride.
7. Issues with pressing share button. Nothing appears on the dashboard when this is pressed.
8. Issues when deleting all video's that Dride takes along time to start recording again.
9. Issues with compression when trying to share by email, WhatsApp etc. The higher the resolution the worst it is. The compression used makes the video clip very blurry and hard to make out things.
10. Issues with video clip length cycle. Even when set to 3 minutes it only records clips for seconds. I have tried this on all time settings resulting in the same.
11. No white/red combo on my end.
12. Timestamp is correct for me.

**Comment by JerryBlack** [comment]
@saoron @BilliTheKid  an update on this new version since my last comment.
There are videos while the vehicle is moving. 
The time is now correct on the recent images.

Thanks

**Comment by saoron** [comment]
Thank you for your help in testing this version, Would appreciate if you could comment on the appropriate thread..

**Comment by ntnlabs** [comment]
I experience the short videos aswell in the original firmware.

**Comment by BilliTheKid** [comment]
The latest app has broken a few bugs that were fixed.

Clip length only 22secs.
Issues with deleting video's and reappearing again.
Compression for sharing video still not fixed.
Pressing sharing does nothing.

Resolution is fine. Still saves.
